sparkle/src/components/Markdown.tsx

```diff
@@ -1,13 +1,41 @@
-import React from "react";
+import React, { useCallback, useState } from "react";
 import ReactMarkdown from "react-markdown";
 
+import { ClipboardCheckIcon, ClipboardIcon, IconButton } from "@sparkle/index";
+
 export function Markdown({
   content,
   className = "",
 }: {
   content: string;
   className?: string;
 }) {
+  function useCopyToClipboard(
+    resetInterval = 2000
+  ): [isCopied: boolean, copy: (d: ClipboardItem) => Promise<boolean>] {
+    const [isCopied, setCopied] = useState(false);
```

Comment:
A bit surprised by this. Did you test it? (console open, to see if there's react messages)
useState should usually be run top of component to which the state belongs. Here you'll have an useState that will only be run when useCopyToClipboard is called.

Maybe there's a use case in which it is legitimate to use it in a different way, in which case glad to learn about it :) Otherwise, IMHO better to avoid calling useState elswhere than top of component, it can lead to weird bugs and is not recommended

Reply:
maybe you could extract CopyToClipboard as a component?

Reply:
TBH @philipperolet I got the same logic as we currently have on the Conversation react component. It is the exact same function. Yes I tested it, and yes I tested it with console opened. 

Is this a blocker for you?

Reply:
as discussed IRL, just extracting the function out of the component would make sense also